libdnf5/base/base.cpp

```diff
@@ -221,7 +233,13 @@ void Base::setup() {
     auto & installroot = config.get_installroot_option();
     installroot.lock("Locked by Base::setup()");
 
-    get_vars()->load(vars_installroot, config.get_varsdir_option().get_value());
+    auto vars = get_vars();
+    const char * arch = nullptr;
+    if (vars->contains("arch")) {
+        arch = vars->get_value("arch").c_str();
+    }
+    init_lib_rpm(arch);
```

Comment:
May I ask you to add a note that when locking of vars will be available we should lock all vars or arch variable.

Are you sure that the order of following lines is correct? I somehow have feeling that we should first load and then init rpm with architecture.
```
init_lib_rpm(arch);
vars->load(vars_installroot, config.get_varsdir_option().get_value());
```

Reply:
The thing is, that during `vars->(load)` the architecture and releasever are also detected (if missing). And this detection needs rpm correctly initialized.

Reply:
I've added a comment on it.